### PR TITLE
Add user profile/level update

### PR DESCRIPTION
**Pull request by henryyang42**
## Add user profile/level update
- can update user profile
- can view others profile
- can change userlevel if authority is satisfied

### CODE REVIEW

**Comment by nickelcheng** [comment]
This PR is ok for me. @CHIAHSINHOU please test this.

**Comment by CHIAHSINHOU** [comment]
When I'm a judge, I change myself to an admin. You should give me 
an error msg. 
I have no idea about this.


When I'm a judge, I change a user to a judge. You should give me an error msg too.

**Comment by henryyang42** [comment]
User level modification error message fixed.
@CHIAHSINHOU  please check if the result is as expected. Thanks.

**Comment by CHIAHSINHOU** [comment]
This PR is good for me.

File: users/static/js/piechartsettings.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ $(function() {
             series: {
                 pie: {
                     show: true,
-                    radius: 180,
+                    radius: 120,
```
Comment (bruce3557) [inline users/static/js/piechartsettings.js]: Is this possible to be a relative value?

File: users/templatetags/profile_filters.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+SOFTWARE.
+'''
+from django import template
+from users.models import User
+from datetime import datetime
+
+register = template.Library()
+
+
+def validate_user(user):
```
Comment (bruce3557) [inline users/templatetags/profile_filters.py]: Is this function duplicated with this PR?
https://github.com/bruce3557/NTHUOJ_web/pull/38/files

File: users/views.py [hunk 3]
```diff
@@ -68,15 +67,56 @@ def submit(request):
...
-    return render(
-        request,
-        'users/profile.html',
-        {'piechart_data': json.dumps(piechart_data)},
-        context_instance=RequestContext(request, processors=[custom_proc]))
+def profile(request, username):
+    try:
+        profile_user = User.objects.get(username=username)
+        piechart_data = []
+        for l in ['WA', 'AC', 'RE', 'TLE', 'MLE', 'OLE', 'Others']:
```
Comment (bruce3557) [inline users/views.py]: Two suggestions:
(1) ```for l in ...``` -> ```for label in ...```
(2) define ```['WA', 'AC', 'RE', 'TLE', 'MLE', 'OLE', 'Others']``` as a constant to enhance extendability

**Comment by bruce3557** [comment]
Great job and left some suggestions to discuss

File: users/templatetags/profile_filters.py (hunk 2 above)
Comment (henryyang42) [inline users/templatetags/profile_filters.py]: Yes, I have extracted this to ``utils.user_info``

File: users/views.py (hunk 3 above)
Comment (henryyang42) [inline users/views.py]: I make use of the constants in problems.models now

**Comment by bruce3557** [comment]
+1, going to merge

**Pull request merged**